### PR TITLE
Fixed #32051 -- Fixed rocket animation on the 'Congrats' page.

### PR DESCRIPTION
**Pull request by thibaudcolas**
This addresses issues reported with the animation in [#32051](https://code.djangoproject.com/ticket/32051), and also includes some code cleanup that should go un-noticed by end users. ~I’m happy to split the code cleanup and actual fixes into separate PRs if this is easier to review.~ Edit: splitting :)

## Code cleanup

There was a lot of unused CSS and SVG, and vendor prefixes that don’t feel relevant anymore (a CPU-heavy animation not running on browsers from 7 years ago is probably a usability win).

Additionally it was hard to extract the SVG to improve how it was defined in design tools – I’ve reworked it so the only styles defined with CSS are the ones used for animations (also makes it easy to display the artwork correctly with animations disabled).

## Rendering fixes

- Removed the Firefox-only `@-moz-document url-prefix()` styles so it gets the same animation as other browsers.
- Allowed the SVG to overflow so it’s no longer cropped when animating, and removed the IE9 workaround `svg:not(:root) { overflow: hidden; }`

## Performance fixes

The animations were already using properties that should be pretty fast and could be GPU-accelerated. They weren’t (at least on my machine), even after applying optimizations (`will-change: transform` didn’t add any layers). I have a feeling this might be due to this being SVG.

I decided to go for a simpler approach and:

- Changed the animations so they each use a single transform. The `scale` transforms were pretty hard to notice
- Added a maximum count to the animations, so they stop after 7s.

## `prefers-reduced-motion`

This was tricky because some of the shapes depend on the animated styles in order to render correctly. I refactored the code so more of the styles are defined directly in the SVG, so I could add the artwork to Sketch, and re-export it without any animation-dependent styles. Then added the media query.

This could have probably be a simpler `animation-play-state: paused`, but it feels safer to support there being no animation styles at all (for people who might disable them via browser extensions). It also makes the code much easier to follow.

---

Tested in latest Chrome, Firefox, Safari on macOS 10.14, as well as IE11. In IE11, only the exhaust__line animation works.

Screen recording of the updated animation, and how it stops (low-FPS GIF, looks smoother in practice):

32051-new-animation

### CODE REVIEW

**Comment by felixxm** [comment]
@thibaudcolas Thanks for this patch :+1: Please move all cleanups unrelated with the main fix (e.g. removing unused CSS) to a separate commit.

**Comment by felixxm** [comment]
@FlipperPA Do you want to take a look?

**Comment by thibaudcolas** [comment]
👍 thank you for taking a look @felixxm, I’ve split this into two commits as requested.

**Comment by pope1ni** [comment]
> ...as well as IE11. In IE11, only the exhaust__line animation works.

We've dropped support for Internet Explorer in the admin site, so I think it is fair to say there is no need to support it here either.
You can drop any hacks that were required for IE.

File: django/views/templates/default_urlconf.html [hunk 1]
```diff
@@ -79,137 +67,51 @@
...
+              transform: translate3d(0, 0, 0);
             }
             100% {
-              -webkit-transform: translate(265px, 253px) scale(2.2) rotate(180deg) skew(-5deg);
-              transform: translate(265px, 253px) scale(2.2) rotate(180deg) skew(-5deg);
+              transform: translate3d(0, 3px, 0);
             }
           }
-          @-webkit-keyframes thurst {
+          @keyframes thurst {
```
Comment (pope1ni) [inline django/views/templates/default_urlconf.html:102]: We could fix the typo while you're here: `thurst` → `thrust`.

Comment (thibaudcolas) [reply django/views/templates/default_urlconf.html:102]: Wow! I have been looking at this code for hours and didn’t even notice. Thank you, will update later today :)

**Comment by thibaudcolas** [comment]
Thank you for the feedback @pope1ni! I’ll update this later today.

I have seen Django’s announcement about dropping support for IE11 on the admin site ([browser support docs](https://docs.djangoproject.com/en/dev/faq/admin/#what-browsers-are-supported-for-using-the-admin)), but wasn’t sure how it applied for things that technically aren’t part of the admin, like this page, Django’s default error pages, and similar "technically not the admin but still somewhat DEBUG-only, developer-focused pages". Are there any official guidelines / has there been discussion about whether those pages follow the same browser support targets or different?

**Comment by thibaudcolas** [comment]
Now updated to fix the typo, and remove the last remnants of IE compatibility code (`display: block` for HTML5 elements).

**Review by felixxm — COMMENTED** [review]
@thibaudcolas Thanks :+1: 

I pushed one small change:
```diff
diff --git a/django/views/templates/default_urlconf.html b/django/views/templates/default_urlconf.html
index b2a5c9dd8b..dc825cf8c2 100644
--- a/django/views/templates/default_urlconf.html
+++ b/django/views/templates/default_urlconf.html
@@ -112,7 +112,7 @@
           main p {
             font-size: 16px;
             line-height: 20px;
-            max-width: 390px;
+            max-width: 410px;
             margin: 15px auto 0;
             color: #888888;
           }
```
it's still much smaller than `max-width` of `h2` (500px) but it removes scroll in some languages, e.g.

- before:
obraz
obraz

- after:
obraz
obraz

**Comment by felixxm** [comment]
@thibaudcolas Welcome aboard :boat:

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Comment by felixxm** [comment]
Merged in 838e115ab159174f759f09d63398e1eed14e4dbe and b807298c666711ba106b06fef05a25073c62b802.

**Comment by thibaudcolas** [comment]
Wonderful, thank you!

**Pull request closed without merging**